Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -51,14 +53,18 @@ public class PythonCompiler : Compiler {
 
     protected override IClassWriter CreateClass(string moduleName, string name, bool isExtern, string fullPrintName,
         List<TypeParameter> typeParameters, TopLevelDecl cls, List<Type> superClasses, IToken tok, ConcreteSyntaxTree wr) {
-      wr.Write("class {0}:", MangleName(name));
-      var methodWriter = wr.NewBlock(open: BlockStyle.Newline, close: BlockStyle.Nothing);
+      var methodWriter = wr.NewBlock(header: $"class {MangleName(name)}:", open: BlockStyle.Newline, close: BlockStyle.Nothing);
+
+      var needsConstructor = cls is TopLevelDeclWithMembers decl && decl.Members.FindAll(m => !m.IsGhost && m is Field && !m.IsStatic).Count > 0;
+      var constructorWriter = needsConstructor
+        ? methodWriter.NewBlock(header: "def  __init__(self):", open: BlockStyle.Newline, close: BlockStyle.Newline)
+        : null;
       if (cls is ClassDecl d) {
-        if (d.Members.FindAll(m => !m.IsGhost).Count == 0) {
+        if (!needsConstructor && d.Members.FindAll(m => !m.IsGhost).Count == 0) {
           methodWriter.WriteLine("pass");
         }
       }
-      return new ClassWriter(this, methodWriter, null);
+      return new ClassWriter(this, constructorWriter: constructorWriter, methodWriter: methodWriter);
```

Comment:
Can't you just put the methodWriter first and the constructorWriter second, so that you don't need to duplicate parameter names?